libpromises/class.c

```diff
@@ -244,11 +242,20 @@ ClassTableIterator *ClassTableIteratorNew(const ClassTable *table, const char *n
 
 Class *ClassTableIteratorNext(ClassTableIterator *iter)
 {
-    void *key_ref = NULL;
-    Class *cls = NULL;
+    MapKeyValue *keyvalue;
 
-    while (RBTreeIteratorNext(iter->iter, &key_ref, (void **)&cls))
+    while ((keyvalue = MapIteratorNext(&iter->iter)) != NULL)
     {
+        Class *cls = keyvalue->value;
+
+        /* Make sure we never store "default" as namespace in the ClassTable,
+         * instead we have always ns==NULL in that case. */
+        CF_ASSERT_FIX(cls->ns == NULL ||
+                      strcmp(cls->ns, "default") != 0,
+                      (cls->ns = NULL),
+                      "Class table contained \"default\" namespace,"
+                      " should never happen!");
```

Comment:
I'm confused about this part. Here you say that you don't store the namespace, but you *do* store the namespace specifically in `ClassTablePut()`, by using `fullname`. Am I missing something?